### DISCUSSION TITLE
Redirect to default FO language when current page language is disabled

### DISCUSSION THREAD

**Discussion by aniszr** [PrestaShop/PrestaShop#32911] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hello,

When you disable all existing languages in BO and just leave the default language enabled, in FO when you refresh your page you get an error message displayed and the browser header displays 404 error !
You need to click on any link in the page so you get a normal display (Refreshing the browser doesn't work!)

See attached screenshot:
![404-error](https://github.com/PrestaShop/PrestaShop/assets/92626242/8039818b-5e66-4036-b0e0-ad0ac892fe07)

Thanks!

### Expected behavior

The page should be displayed normally and displayed with default language (chosen in shop installation)

![shop80X](https://github.com/PrestaShop/PrestaShop/assets/92626242/8900fb9c-e86f-4ec8-a510-3575b5ee14a7)

### Steps to reproduce

1. Open an FO page & check languages in drop-down menu
1. Go to BO > International > Localization > Languages
2. Disable all languages you have and just leave the default one
3. Back to FO > Refresh the browser
4. See error displayed
5. Click on any link in the page > error is gone

See attached videorecord:

https://github.com/PrestaShop/PrestaShop/assets/92626242/409aa65c-7469-46c3-8f9e-ebee7e705670

### PrestaShop version(s) where the bug happened

1.7.8.9 & 8.0.4 & 8.1.x

### PHP version(s) where the bug happened

7.4

**Comment by sallemiines** [comment]
Hello @aniszr 

Thanx for this issue ! 
I  reproduce your issue with ps  1787 ,1789 , 804 , 81x , 

https://github.com/PrestaShop/PrestaShop/assets/31663369/c1f369fe-f73b-47d4-bdee-63a6c8d52536

it's an old bug 
Thank you  !

**Comment by dnkhack** [comment]
@sallemiines @aniszr 
My apologies. But, after you both, disabled the language, you left the page with the old URL with the selected language on the front office. This is not an issue - the page there no longer exists with that suffix.
Just open the home page of your website. And add the suffix "foo" and you will get 404 - it is the same with your issue.
`https://your.site/foo`
For instance, it is the same for modules. You are on some module page - A blog post. If you disable the module - update the page will get 404. Where in this case does PS have to redirect? Nowhere - it is just the wrong URL!  
![image](https://github.com/PrestaShop/PrestaShop/assets/35419462/d37b4eb6-b35f-4ec0-ad27-51f498c13223)
![image](https://github.com/PrestaShop/PrestaShop/assets/35419462/2c5ebb0b-08f6-4f3d-9b65-cccbb249b2ac)

**Comment by aniszr** [comment]
Ping @MatShir @RosaBenouamer Is this the expected behavior? or PS should automatically change the language in URL by the default language left in BO?

Thanks in advance.

**Comment by MatShir** [comment]
It is the expected behavior, the links are not available anymore. But I'm wondering if a redirection can be added, like @aniszr suggested 🤔 
Anyway, not a bug but rather a feature request :)

**Reply by dnkhack** [reply]
@MatShir hello.  I believe this is no need for any improvements. Because if you make that "trick" on a product or CMS page then PS redirects to the same page with language by default! In this case, it was the home page! What behavior are you expect? Redirect to the home page with lang by default? But in this case, all 404 will be redirected to the home page with lang by default. 
If we talking about page index in search services like Google, then here customers can use paid modules from PS Addons or try to find something free. If to write a code to Dispatcher for that case, then 
- the conditions of disabled or deleted lang
- the conditions of a multishop 
- the conditions of a subdomain, a subfolder
ok redirect
For what those tons of code?

**Reply by MatShir** [reply]
I think you're right! Hard work for little value, and that's a very unlikely scenario.
